ibis/expr/operations/core.py

```diff
@@ -114,14 +117,80 @@ def flat_args(self):
 
 @public
 class ValueOp(Node):
+    # columnar if any of the arguments is column-like
+    output_shape = rlz.shape_like("args")
+
+    # TODO(kszucs): we may want to enable this by default instead of
```

Comment:
Since operations are now immutable we can materialise `output_shape` and `output_dtype` on construction. Should experiment with this.

Eventually we should materialise `output_type` for all `Node` objects on construction.

Reply:
Updated.